bookwyrm/models/book.py

```diff
@@ -491,6 +492,12 @@ def hyphenated_isbn13(self):
         """generate the hyphenated version of the ISBN-13"""
         return hyphenator.hyphenate(self.isbn_13)
 
+    @property
+    def average_rating(self, user):
+        """generate average rating of a book"""
+        reviews = Review.privacy_filter(user).filter(book__parent_work__editions=self)
+        return reviews.aggregate(Avg("rating"))["rating__avg"]
```

Comment:
`@property` functions cannot take arguments as mentioned [here](https://pylint.pycqa.org/en/latest/user_guide/messages/refactor/property-with-parameters.html)

However the [original implementation](https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/views/books/books.py#L101) is on run a `reviews` value that has the class method `privacy_filter` run on it _**before**_ the rating calculation is done as shown [here](https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/views/books/books.py#L68-L70)

Should we remove the `privacy_filter` call or switch to making this a function on the `Edition` model or a `classmethod`?

Reply:
Also, other calculations of the `rating` use the `user` argument, strengthening the idea that this should be either a method on the model or a class method. Examples are [here](https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/views/annual_summary.py#L74-L79), and [here](https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/templatetags/rating_tags.py#L19-L23). The latter also looks for ratings greater than 0.

The calculation [here](https://github.com/bookwyrm-social/bookwyrm/blob/main/bookwyrm/preview_images.py#L455-L459) doesn't use the `user` argument.

Another idea could be to have different methods that keep the behaviour of the methods mentioned but have them be on the `Edition` model rather than in the different places they are in right now.

Reply:
Using a `classmethod` seems like a solid choice to me